### PR TITLE
NTV-280: Cannot edit a pledge with a sold out reward on a project with add-ons

### PR DESCRIPTION
**Pull request by Arkariang**
# 📲 What

- Be able to edit your addOns selection, when the base reward is sold out or not available.

# 👀 See
- Pledge to limited reward with allowed addons, but do not choose any addOn -> Choose another reward, select same reward, add addOn to pledge -> Choose another reward, select same reward, modify addOns selection

https://user-images.githubusercontent.com/4083656/141998621-a06e1e3d-1cec-4133-81ec-f8e4feedd94b.mp4

| --- | --- |
|  |  |

# 📋 QA
- go to the project https://staging.kickstarter.com/projects/jgfranzen/ik-fantasy-the-black-keep-game-hells-gate-dungeon-box .
- Pledge to limited reward with allowed addons, but do not choose any addOn -> Choose another reward, select same reward, add addOn to pledge -> Choose another reward, select same reward, modify addOns selection

# Story 📖

[NTV-280](https://kickstarter.atlassian.net/browse/NTV-280)

### CODE REVIEW

**Review by leighdouglas — APPROVED** [review]
lgtm!

File: app/src/test/java/com/kickstarter/libs/utils/RewardViewUtilsTest.kt [hunk 1]
```diff
@@ -42,6 +42,20 @@ class RewardViewUtilsTest : KSRobolectricTestCase() {
         assertEquals(R.string.Continue, RewardViewUtils.pledgeButtonText(backedProjectRwHasAvailableNotBackedAddOns, backedRwAvailableAddOns))
     }
 
+    /**
+     * Given a backedProject the backed reward
+     * has available addOns
+     * when not backed addOns
```
Comment (iambchan) [inline app/src/test/java/com/kickstarter/libs/utils/RewardViewUtilsTest.kt:48]: what happens on android if you have backed add-ons in your original pledge?

File: app/src/main/java/com/kickstarter/viewmodels/RewardViewHolderViewModel.kt [hunk 2]
```diff
@@ -392,8 +392,8 @@ interface RewardViewHolderViewModel {
         /**
          * Use cases for enabling/disabling access to launch the next fragment
          * - If the selected reward has no addOns the CTA button will be enable if available
-         * - If the previously selected reward has addOns but not BackedAddOns CTA button available is reward available
-         * - If the previously selected reward has add-ons and has Backed addOns, CTA button available (they still can update the addOns selection)
+         * - If selecting other reward CTA button available if reward available
```
Comment (iambchan) [inline app/src/main/java/com/kickstarter/viewmodels/RewardViewHolderViewModel.kt:395]: does it still work if you're selecting the same sold out reward but just changing the add-ons?
The use case is, backers usually don't want to lose their early bird reward that's sold out but they want to add add-ons as creators add more of them throughout the campaign

File: app/src/test/java/com/kickstarter/libs/utils/RewardViewUtilsTest.kt (hunk 1 above)
Comment (Arkariang) [reply app/src/test/java/com/kickstarter/libs/utils/RewardViewUtilsTest.kt:48]: You can see the behaviour on the video from second 20 onwards

File: app/src/main/java/com/kickstarter/viewmodels/RewardViewHolderViewModel.kt (hunk 2 above)
Comment (Arkariang) [reply app/src/main/java/com/kickstarter/viewmodels/RewardViewHolderViewModel.kt:395]: if selecting the same reward the button will be available for the user to be able to navigate to the AddOns Screen

Comment (iambchan) [reply app/src/main/java/com/kickstarter/viewmodels/RewardViewHolderViewModel.kt:395]: thanks for confirming!

File: app/src/test/java/com/kickstarter/libs/utils/RewardViewUtilsTest.kt (hunk 1 above)
Comment (iambchan) [reply app/src/test/java/com/kickstarter/libs/utils/RewardViewUtilsTest.kt:48]: ty! I see it!

**Review by iambchan — APPROVED** [review]
🥳 thanks for fixing this!

**Pull request merged**